qemu/tests/nic_hotplug.py

```diff
@@ -140,12 +140,21 @@ def run_sub_test(params, plug_tag):
                  if needed
         """
         sub_type = params.get("sub_type_%s" % plug_tag)
-        if sub_type:
-            context_msg = "Running sub test '%s' %s"
-            error_context.context(context_msg % (sub_type, plug_tag),
-                                  logging.info)
-            utils_test.run_virt_sub_test(test, params, env, sub_type)
-            return (sub_type == "shutdown")
+        login_timeout = params.get_numeric("login_timeout", 360)
+        session = vm.wait_for_serial_login(timeout=login_timeout)
+        shutdown_method = params.get("shutdown_method", "shell")
+        shutdown_command = params.get("shutdown_command")
+        if sub_type == "boot":
+            logging.info("Running sub test '%s' %s" % (sub_type, plug_tag))
+            if params.get("reboot_method"):
+                vm.reboot(session, params["reboot_method"], 0,
+                          login_timeout, serial=True)
+        elif sub_type == "shutdown":
+            logging.info("Running sub test '%s' %s" % (sub_type, plug_tag))
+            if shutdown_method == "shell":
```

Comment:
if the `shutdown_method` is not shell, do we need to send `system_powerdown` to qmp?

Reply:
Hi @PaulYuuu Because I restricted "shutdown_method = shell" in cfg, so I think the latter situation can be ignored. Please correct me if I'm wrong.

Reply:
`shutdown_method = params.get("shutdown_method", "shell")`, so we can define it as other. This time I can accept it, please pay attention to this in the future.